app/sign.js

```diff
@@ -118,19 +129,23 @@ const handleSignUtxo = function(recoveryRequest, key, skipConfirm) {
 
     console.log(`Signing input ${ i + 1 } of ${ recoveryRequest.inputs.length } with ${ derivedHDNode.neutered().toBase58() } (${ input.chainPath })`);
 
-    if (isBech32) {
-      const witnessScript = Buffer.from(input.witnessScript, 'hex');
-      const witnessScriptHash = utxoLib.crypto.sha256(witnessScript);
-      const prevOutScript = utxoLib.script.witnessScriptHash.output.encode(witnessScriptHash);
-      txBuilder.sign(i, derivedHDNode.keyPair, prevOutScript, utxoLib.Transaction.SIGHASH_ALL, input.amount, witnessScript);
-    } else {
+    if (isBCH) {
```

Comment:
You may be able to simplify the conditional nesting here by returning the `txBuilder` inline (as it looks like you aren't doing much to the txBuilder after this conditional block. ie:

```javascript
// Handle BCH recovery requests
if (recoveryRequest.coin === 'bch' || recoveryRequest.coin === 'tbch') {
  const redeemScript = new Buffer(input.redeemScript, 'hex');
  txBuilder.sign(i, derivedHDNode.keyPair, redeemScript, utxoLib.Transaction.SIGHASH_BITCOINCASHBIP143 | utxoLib.Transaction.SIGHASH_ALL, input.amount);
  return txBuilder.build().toHex();
}

// Handle Bech32 requests
if (!input.redeemScript){
  const witnessScript = Buffer.from(input.witnessScript, 'hex');
  const witnessScriptHash = utxoLib.crypto.sha256(witnessScript);
  const prevOutScript = utxoLib.script.witnessScriptHash.output.encode(witnessScriptHash);
  txBuilder.sign(i, derivedHDNode.keyPair, prevOutScript, utxoLib.Transaction.SIGHASH_ALL, input.amount, witnessScript);
  return txBuilder.build().toHex();
}

// Handle segwit requests
const redeemScript = new Buffer(input.redeemScript, 'hex');
if (!!input.witnessScript){
  const witnessScript = new Buffer(input.witnessScript, 'hex');
  txBuilder.sign(i, derivedHDNode.keyPair, redeemScript, utxoLib.Transaction.SIGHASH_ALL, input.amount, witnessScript)
  return txBuilder.build().toHex();
}

// Handle all other requests
txBuilder.sign(i, derivedHDNode.keyPair, redeemScript, utxoLib.Transaction.SIGHASH_ALL, input.amount);
return txBuilder.build().toHex();
```

Less nesting, easier to read and maintain ¯\_(ツ)_/¯

Reply:
its not quite that simple because all of that is inside of a for loop. we need to sign for each input, so we can't return mid-script like you have above.